### PR TITLE
Fix setting innerHTML on non-HTMLnamespaced elements

### PR DESCRIPTION
**Pull request by TimvdLippe**
Part of https://github.com/webcomponents/shadydom/issues/228 and required for https://github.com/webcomponents/shadydom/pull/229
The patch for `innerHTML` on Safari 9 called `createElement`, but for non-HTMLnamespaced elements this should be `createElementNS` with the `this.namespaceURI` instead.

### CODE REVIEW

File: src/Patch/Element.js [hunk 1]
```diff
@@ -100,8 +100,14 @@ export default function(internals) {
...
           /** @type {!Node} */
-          const rawElement = Native.Document_createElement.call(document,
-            this.localName);
+          let rawElement;
+          if (this.namespaceURI === document.namespaceURI) {
+            rawElement = Native.Document_createElement.call(document,
+              this.localName);
+          } else {
+            rawElement = Native.Document_createElementNS.call(document,
+              this.namespaceURI, this.localName);
```
Comment (bicknellr) [inline src/Patch/Element.js]: You can remove the if-else and use this line by itself: `createElementNS` can create elements in the HTML namespace too. (i.e. `document.createElement('button')` and `document.createElementNS('http://www.w3.org/1999/xhtml', 'button')` will both produce a regular HTML `<button>`.)

**Comment by TimvdLippe** [comment]
@bicknellr Apparently Edge 14 `svg.innerHTML` is broken by itself, so we have no other option than to ignore the test :cry:

**Comment by bicknellr** [comment]
What exactly is the behavior of Edge 14? Instead of disabling the test based on the UA string, could you check the behavior before the polyfill is loaded and decide from that? For example, in `insertAdjacentHTML`:

https://github.com/webcomponents/custom-elements/blob/fe10dc8a5b1936d839181fa65ded5903281d6c2b/tests/html/Element/insertAdjacentHTML.html#L6-L14

https://github.com/webcomponents/custom-elements/blob/fe10dc8a5b1936d839181fa65ded5903281d6c2b/tests/html/Element/insertAdjacentHTML.html#L171-L181

**Comment by TimvdLippe** [comment]
@bicknellr Ah cool, yes that is better. Updated :+1:

File: tests/html/Element/innerHTML.html [hunk 2]
```diff
@@ -165,14 +173,25 @@
...
+
+  const canSVGInnerHTML = (function() {
+    try {
+      // Edge 14 svg.innerHTML is broken even without the polyfill
+      const svg = NATIVE.createElementNS.call(document, SVG_NAMESPACE, 'svg');
+      NATIVE.innerHTML.call(svg, circleText);
+      return svg.querySelector('circle') instanceof SVGCircleElement;
+    } catch (e) {
+      return false;
+    }
```
Comment (bicknellr) [inline tests/html/Element/innerHTML.html]: Is this expected to throw in some cases? If not, you don't need to try-catch, IMO.

File: tests/html/Element/innerHTML.html [hunk 3]
```diff
@@ -163,6 +172,26 @@
...
+  const canSVGInnerHTML = (function() {
+    // Edge 14 svg.innerHTML is broken even without the polyfill
+    const svg = NATIVE.createElementNS.call(document, SVG_NAMESPACE, 'svg');
+    NATIVE.innerHTML.call(svg, circleText);
+    return svg.querySelector('circle') instanceof SVGCircleElement;
+  })();
+
+  test('svg innerHTML', function() {
+    if (!canSVGInnerHTML) {
+      this.skip();
```
Comment (bicknellr) [inline tests/html/Element/innerHTML.html]: Oh, this is nice to know about. I think there are a few other places in the tests where I ended up switching out the test function itself:
```
const canDoTheThing = ... some test ... ;
(canDoTheThing ? test : test.skip)('test name', function() { ...
```
Calling `this.skip()` in them would be much cleaner.

**Pull request merged**